### PR TITLE
Adds Paul'st article about auth0 simulator and nav option

### PR DESCRIPTION
**Pull request by jorgelainfiesta**
## Motivation

Announce Auth0 simulator

## Approach

Paul wrote article, I edited and proofed with Olson. Also added Auth0 simulator link under Open Source drop down

### CODE REVIEW

File: src/blog/2022-01-13-auth0-simulator.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+```
+
+## Local Development
+
+The goal of `@simulacrum/auth0-simulator` is to behave just like the real Auth0 server does. That means that libraries like `@auth0/auth0-react` should not notice any difference so that you can keep using your regular code and expect it to work without any adjustment when you switch to the real deal. The following snippet would work just as well using the Auth0 simulator and the production environment.
+
+```ts
+import { useAuth0 } from '@auth0/auth0-react';
+
+<Button onClick={() => loginWithRedirect()}>  Log in   </Button>;
```
Comment (dagda1) [inline src/blog/2022-01-13-auth0-simulator.md]:
```suggestion
import { useAuth0 } from "@auth0/auth0-react";

const LoginButton = () => {
  const { loginWithRedirect } = useAuth0();

  return <button onClick={() => loginWithRedirect()}>Log In</button>;
};
```

**Review by dagda1 — CHANGES_REQUESTED** [review]
Nice work!  Just one update to a code example

**Comment by jorgelainfiesta** [comment]
@dagda1 I committed your change, thanks! can you do a final check so we can publish it? thanks!

**Comment by dagda1** [comment]
> @dagda1 I committed your change, thanks! can you do a final check so we can publish it? thanks!

I think it is good to go.  When are you thinking of publishing this?  I have a couple of PRs that I would like to get merged in the auth0 simulator.

It would be good to have those merged in case anyone feels inspired to give it a go after reading the blog post.

**Comment by jorgelainfiesta** [comment]
I can publish it whenever you feel it's ready

File: src/blog/2022-01-13-auth0-simulator.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+Packages like [mock-jwks](https://www.npmjs.com/package/mock-jwks) help, but it’s still code that needs to be maintained, and as we all know, the less code we write, the fewer problems we have.
+A third solution is creating phony Auth0 accounts for different environments, but that too is a maintainability nightmare. The accounts on each environment must now be kept in sync, which is unlikely to happen.
+
+## Solution
+
+However, the unworkable “solutions” presented above also point to the principle that will guide our workable one: namely that we should always strive to have the same code running in every environment. The only difference should be the configuration.
+For example, in production we might have this auth0 configuration:
+
+```JSON
```
Comment (cowboyd) [inline src/blog/2022-01-13-auth0-simulator.md:80]: It looks like this is not valid JSON. 

I think there are quotes missing around the "domain" key.

File: src/blog/2022-01-13-auth0-simulator.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+import { createClient } from '@simulacrum/client';
+import auth0Config from '../../cypress.env.json';
+
+describe('log in', () => {
+  it('should get token without signing in', () => {
+    cy.createSimulation(auth0Config)
+      .visit('/')
+      .contains('Log out')
+      .should('not.exist')
+      .given()
```
Comment (cowboyd) [inline src/blog/2022-01-13-auth0-simulator.md]: `given()` without any context doesn't really read well.

```ts
given({
  email: 'bob@gmail.com'
});
```

makes sense, but just
```ts
given()
```
is a bit weird. Not sure what to do? Thoughts?

File: src/blog/2022-01-13-auth0-simulator.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      }
+
+      resolve(decoded);
+    });
+  });
+};
+```
+
+Packages like [mock-jwks](https://www.npmjs.com/package/mock-jwks) help, but it’s still code that needs to be maintained, and as we all know, the less code we write, the fewer problems we have.
+A third solution is creating phony Auth0 accounts for different environments, but that too is a maintainability nightmare. The accounts on each environment must now be kept in sync, which is unlikely to happen.
```
Comment (cowboyd) [inline src/blog/2022-01-13-auth0-simulator.md]: Are we underselling the pain associated with this setup here? Basically, you have a single "fake" online auth0 tenant for testing, which is a global dumping ground for anybody that happens to be doing testing in your organization at any given moment. There are so many cons to this very common approach that I feel like we should say them out loud:

1. You are at the mercy of Auth0 to provision you each testing tenant
1. You cannot test if Auth0 is unavailable or your are developing "behind the firewall"
1. You are still subject to rate limits of however many testers happen to be using it. When that rate limit is breached, _all_ tests in your whole organization are blocked.
3. The test accounts share state with everyone who happens to be running a test _at the same time_, and so every test has a hidden dependency on _every other test_.

**Review by cowboyd — COMMENTED** [review]
Looks great! some minor comments inline.

File: src/blog/2022-01-13-auth0-simulator.md (hunk 3 above)
Comment (jorgelainfiesta) [reply src/blog/2022-01-13-auth0-simulator.md]: We can add the email as you suggest, I don't have any strong opinion

**Review by cowboyd — APPROVED** [review]

**Review by dagda1 — APPROVED** [review]
👍

**Comment by taras** [comment]
@jorgelainfiesta @dagda1 the link in the conclusion of the blog post links to simulacrum repo. Should it point to https://github.com/thefrontside/simulacrum/tree/v0/packages/auth0?

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/thefrontside/simulacrum">simulacrum/packages/auth0 at v0 · thefrontside/simulacrum</a></strong></div><div>A simulation platform for use during testing, during development and for high-fidelity application previews - simulacrum/packages/auth0 at v0 · thefrontside/simulacrum</div></blockquote>

**Comment by dagda1** [comment]
> @jorgelainfiesta @dagda1 the link in the conclusion of the blog post links to simulacrum repo. Should it point to https://github.com/thefrontside/simulacrum/tree/v0/packages/auth0?
> 
> >  GitHub**[simulacrum/packages/auth0 at v0 · thefrontside/simulacrum](https://github.com/thefrontside/simulacrum)**A simulation platform for use during testing, during development and for high-fidelity application previews - simulacrum/packages/auth0 at v0 · thefrontside/simulacrum

@taras yes, good spot, it should point to the auth0 package.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/thefrontside/simulacrum">simulacrum/packages/auth0 at v0 · thefrontside/simulacrum</a></strong></div><div>A simulation platform for use during testing, during development and for high-fidelity application previews - simulacrum/packages/auth0 at v0 · thefrontside/simulacrum</div></blockquote>

**Pull request merged**